### PR TITLE
Jump median fix

### PR DESCRIPTION
**Pull request by mwregan2**
This is a fix for GitHub issue #3958 , Incorrect median calculation in Jump Detection. It has the fix and a few more unit tests that recreated the error and tested other median calculations where the number of samples and bad frames are varied.

### CODE REVIEW

**Comment by jdavies-st** [comment]
Hi Mike, it looks like Pycharm produces a bunch of temporary files in your working dir.  To avoid having these tracked by `git`, you can add them to your global git ignore file.

https://help.github.com/en/articles/ignoring-files#create-a-global-gitignore

That way you don't have to worry about them in whatever repo you are contributing.  👍

**Comment by mwregan2** [comment]
Jim,
  Thanks, I know about the problem but slipped up and they got added to the commit. I thought I removed them from the pull request.
  I had put them in the .gitignore but then that got into a pull request. So, I think that if I add them to the .gitignore and then add .gitignore to .gitignore, I may be able to avoid this in the future.
Mike

File: jwst/jump/tests/test_twopoint_difference.py [hunk 1]
```diff
@@ -474,6 +572,69 @@ def test_10grps_satat8_crsat3and6(setup_cube):
...
+    nframes=1
+    data[0, 0, 100, 100] = 0
+    data[0, 1, 100, 100] = 4500
+    data[0, 2, 100, 100] = 9100
+    data[0, 3, 100, 100] = 13800
+    data[0, 4, 100, 100] = 18600
+    data[0, 5, 100, 100] = 40000 #CR
+    data[0, 6, 100, 100] = 44850
+    data[0, 7, 100, 100] = 49900
+    data[0, 8:10, 100, 100] = 60000
```
Comment (dmggh) [inline jwst/jump/tests/test_twopoint_difference.py]: Similar comment here.

File: jwst/jump/twopoint_difference.py [hunk 2]
```diff
@@ -187,28 +186,27 @@ def get_clipped_median(num_differences, diffs_to_ignore, differences, sorted_ind
...
+        row, col = np.indices(diffs_to_ignore.shape)
+        pixel_med_index = sorted_index[row, col, (num_differences - (diffs_to_ignore[row, col] + 1))//2]
+        pixel_med_diff = differences[row, col, pixel_med_index]
+        # Get the row and column of each pixel.)
 
         # In addition, decrease the index by 1 for every two diffs_to_ignore,
         # these will be saturated values in this case
-        pixel_med_diff = differences[row, col, pixel_med_index - (diffs_to_ignore / 2).astype(int)]
         # For pixels with an even number of differences the median is the mean of the two central values
         # So we need to get the
```
Comment (dmggh) [inline jwst/jump/twopoint_difference.py]: Incomplete comment.    Also, does this work if there are 0 pixels with an even number of differences ?

Comment (mwregan2) [reply jwst/jump/twopoint_difference.py]: David,
  I reorganized the comments to better explain what is going on.
Thanks
Mike

File: jwst/jump/tests/test_twopoint_difference.py (hunk 1 above)
Comment (mwregan2) [reply jwst/jump/tests/test_twopoint_difference.py]: David,
  I fixed these three index problems.
Mike

**Comment by mwregan2** [comment]
I removed all the *.txt files.

**Comment by mwregan2** [comment]
> The `all*.txt` files need to be removed from this PR.

I have removed the *.txt files

**Comment by jdavies-st** [comment]
Thanks @mwregan2.  Your changes look good.  The tests are still failing though.  See details in the links below.

https://travis-ci.org/spacetelescope/jwst/jobs/582699281#L939-L1005

**Comment by mwregan2** [comment]
I messed up two of the tests. I have fixed them now. @jdavies-st

**Comment by mwregan2** [comment]
The tests are failing but it seems to be a global issue not one on the unit tests. Travis is having a problem downloading python 3.6.8 and I can't figure out what Jenkins is complaining about.

**Comment by stscieisenhamer** [comment]
@mwregan2 The one travis error is a flake8 error: 

jwst/jump/twopoint_difference.py:205:111: W291 trailing whitespace

Once that is handled, all should be good.

**Pull request merged**